pkg/controller/endpointslice/reconciler_test.go

```diff
@@ -430,6 +483,50 @@ func TestReconcileEndpointSlicesUpdating(t *testing.T) {
 	expectUnorderedSlicesWithLengths(t, fetchEndpointSlices(t, client, namespace), []int{100, 100, 50})
 }
 
+// In some cases, such as service labels updates, all slices for that service will require a change
+// This test ensures that we are updating those slices and not calling create + delete for each
+func TestReconcileEndpointSlicesServicesLabelsUpdating(t *testing.T) {
```

Comment:
Can you add a similar test (or test case) that covers the same scenario but with a different reserved label just to make sure we're not updating EndpointSlices in that case?